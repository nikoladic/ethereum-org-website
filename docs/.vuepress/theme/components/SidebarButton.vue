<template>
  <div class="sidebar-button">
    <a href="#" @click="$emit('toggle-sidebar')" class="flex">
      <icon name="menu" />
    </a>
    <router-link :to="$localePath" class="home-link">
      <span
        ref="siteName"
        class="site-name"
        v-if="$siteTitle"
        :class="{ 'can-hide': $site.themeConfig.logo }"
        >{{ $siteTitle }}</span
      >
    </router-link>
  </div>
</template>

<style lang="stylus">
@import '../styles/config.styl'

.sidebar-button
  display none
  margin-left: -0.4em

  svg
    cursor: pointer
    margin-right 1em
    path
      fill $textColor

@media (max-width $breakS)
  .sidebar-button
    display flex
    align-items center
</style>
