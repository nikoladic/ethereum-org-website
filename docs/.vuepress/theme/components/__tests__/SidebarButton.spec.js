import { shallowMount } from '@vue/test-utils'
import SidebarButton from "@/theme/components/SidebarButton"

describe('SidebarButton', () => {
  test('render without site title', () => {
    const wrapper = shallowMount(SidebarButton, {
      stubs: ['router-link']
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  test('render with site title', () => {
    const wrapper = shallowMount(SidebarButton, {
      mocks: {
        $site: {
          themeConfig: {
            logo: "./eth-diamond"
          }
        },
        $siteTitle: "Ethereum"
      },
      stubs: ['router-link']
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})

