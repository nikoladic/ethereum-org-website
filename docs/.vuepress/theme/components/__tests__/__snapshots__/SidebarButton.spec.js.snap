// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarButton render with site title 1`] = `
<div
  class="sidebar-button"
>
  <svg
    aria-hidden="true"
    class="icon"
    height="20px"
    role="img"
    viewBox="0 0 25 20"
    width="25px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      Hamburger menu button
    </title>
     
    <g
      transform="translate(-22.000000, -2.000000)"
    >
      <g
        transform="translate(22.000000, 2.545974)"
      >
        <path
          d="M24.7714844,0.5 L0,0.5"
        />
         
        <path
          d="M24.7714844,9.72701303 L0,9.72701303"
        />
         
        <path
          d="M24.7714844,18.9540261 L0,18.9540261"
        />
      </g>
    </g>
  </svg>
   
  <router-link-stub
    class="home-link"
  >
    <span
      class="site-name can-hide"
    >
      Ethereum
    </span>
  </router-link-stub>
</div>
`;

exports[`SidebarButton render without site title 1`] = `
<div
  class="sidebar-button"
>
  <svg
    aria-hidden="true"
    class="icon"
    height="20px"
    role="img"
    viewBox="0 0 25 20"
    width="25px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      Hamburger menu button
    </title>
     
    <g
      transform="translate(-22.000000, -2.000000)"
    >
      <g
        transform="translate(22.000000, 2.545974)"
      >
        <path
          d="M24.7714844,0.5 L0,0.5"
        />
         
        <path
          d="M24.7714844,9.72701303 L0,9.72701303"
        />
         
        <path
          d="M24.7714844,18.9540261 L0,18.9540261"
        />
      </g>
    </g>
  </svg>
   
  <router-link-stub
    class="home-link"
  >
    <!---->
  </router-link-stub>
</div>
`;
